lib/rules/sort-imports.js

```diff
@@ -43,12 +45,81 @@ module.exports = {
     },
 
     create: function(context) {
-
         const configuration = context.options[0] || {},
             ignoreCase = configuration.ignoreCase || false,
             ignoreMemberSort = configuration.ignoreMemberSort || false,
-            memberSyntaxSortOrder = configuration.memberSyntaxSortOrder || ["none", "all", "multiple", "single"];
-        let previousDeclaration = null;
+            memberSyntaxSortOrder = configuration.memberSyntaxSortOrder || ["none", "all", "multiple", "single"],
+            sourceCode = context.getSourceCode(),
+            reactRegExp = /\/?[^\/]*?react[^\/]*?$/ig;
```

Comment:
We can't favor specific libraries or frameworks in core rules. You'll want to make this more generic, maybe something like `firstImportsRegExp` and let the user configure it. (We also can't have a default value for this, for the same reason.)